boudicca.base/publisher-event-html/src/test/kotlin/base/boudicca/publisher/event/html/PublisherHtmlApplicationE2ETests.kt

```diff
@@ -0,0 +1,122 @@
+import base.boudicca.publisher.event.html.testdata.E2EGeneralTestData
+import base.boudicca.publisher.event.html.testdata.E2ESingleEventTestData
+import base.boudicca.publisher.event.html.testdata.E2ESingleEventWithoutURL
+import base.boudicca.publisher.event.html.util.SnapshotHandler.SnapshotHandler
+import com.microsoft.playwright.assertions.PlaywrightAssertions.assertThat
+import org.junit.jupiter.api.Tag
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ArgumentsSource
+import org.mockito.kotlin.any
+import org.mockito.kotlin.whenever
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.boot.test.web.server.LocalServerPort
+import org.springframework.test.context.junit.jupiter.SpringExtension
+
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ExtendWith(SpringExtension::class)
+@Tag("playwright")
+class PublisherHtmlApplicationE2ETests: E2ETestFixture() {
+
+  @LocalServerPort
+  private val port = 0
+
+  @MockBean
+  lateinit var searchServiceCaller: SearchServiceCaller
+
+  @ParameterizedTest
+  @ArgumentsSource(E2EGeneralTestData::class)
+  fun pageShouldHaveCorrectTitle(
+    events: List<Event>,
+    filters: Map<String, List<String>>
+  ) {
+    setupSearchServiceCaller(events, filters)
+
+    page.navigate("http://localhost:$port/")
+    assertThat(page).hasTitle("Boudicca")
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(E2ESingleEventTestData::class)
+  fun eventShouldBeVisibleInViewport(
+    events: List<Event>,
+    filters: Map<String, List<String>>
+  ) {
+    setupSearchServiceCaller(events, filters)
+
+    page.navigate("http://localhost:$port/")
+
+    assertThat(page.locator(".event")).isInViewport()
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(E2ESingleEventTestData::class)
+  fun eventDetailsShouldBeCorrect(
+    events: List<Event>,
+    filters: Map<String, List<String>>
+  ) {
+    setupSearchServiceCaller(events, filters)
+
+    page.navigate("http://localhost:$port/")
+
+    assertThat(page.locator(".event-details").first()).containsText("01.01.2023 um 13:00 Uhr")
+    assertThat(page.locator(".event-details").last()).containsText("Theater, Linz")
+    assertThat(page.locator(".event-title")).equals("Musical Event in Innenstadt")
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(E2ESingleEventWithoutURL::class)
+  fun eventImageShouldBeCorrect(
+    events: List<Event>,
+    filters: Map<String, List<String>>
+  ) {
+    setupSearchServiceCaller(events, filters)
+
+    val snapshotHandler = SnapshotHandler("eventImageShouldBeCorrect.snapshot", "snapshot")
+
+    page.navigate("http://localhost:$port/")
+
+    val image = page.locator(".event-image").innerHTML()
+
+    if (!snapshotHandler.exists()) {
+      snapshotHandler.save(image)
+    } else {
+      assert(image == snapshotHandler.read())
```

Comment:
i think those are the wrong assert calls? those seem to be java asserts, not junit or playwright asserts. i am not sure they actually work as they are supposed to

Reply:
yes, you are right. I'll replace those with assertion from Junit, since Playwright only provide locator assertion.